packages/core/src/Machine.ts

```diff
@@ -69,7 +55,7 @@ export function createMachine<
   // Ensure that only the first overload matches models, and prevent
   // accidental inference of the model as the `TContext` (which leads to cryptic errors)
   config: TContext extends Model<any, any, any, any>
-    ? never
+    ? 'Passing typeof model to the first generic of createMachine is no longer supported - use model.createMachine instead'
```

Comment:
@Andarist will this work as intended?

Reply:
Also, there are now only two overloads left in this file. They look very similar in signature - can we condense them into... only... ONE OVERLOAD...?

Reply:
```ts
export function createMachine<
  TContext,
  TEvent extends EventObject = AnyEventObject,
  TTypestate extends Typestate<TContext> = { value: any; context: TContext }
>(
  // Ensure that only the first overload matches models, and prevent
  // accidental inference of the model as the `TContext` (which leads to cryptic errors)
  config: TContext extends Model<any, any, any, any>
    ? 'Passing typeof model to the first generic of createMachine is no longer supported - use model.createMachine instead'
    : MachineConfig<TContext, any, TEvent>,
  options?: Partial<MachineOptions<TContext, TEvent>>
): StateMachine<TContext, any, TEvent, TTypestate>;
export function createMachine<
  TContext,
  TEvent extends EventObject = AnyEventObject,
  TTypestate extends Typestate<TContext> = { value: any; context: TContext }
>(
  config: MachineConfig<TContext, any, TEvent>,
  options?: Partial<MachineOptions<TContext, TEvent>>
): StateMachine<TContext, any, TEvent, TTypestate> {
  return new StateNode<TContext, any, TEvent, TTypestate>(
    config,
    options
  ) as StateMachine<TContext, any, TEvent, TTypestate>;
}
```

Reply:
Note that `createMachine` has only a single overload here now - the second thing is like a private declaration. If possible you could still consolidate both though